### PR TITLE
Add last documentation comments for methods used to run 'fs ls ...' in Alluxio

### PR DESCRIPTION
**Pull request by abenaru-zz**
# Pull Request Merge Checklist
           
Task backlog identifier related to this PR: [277](https://github.com/s1mbi0se/simbiose_knowledge_base/issues/277)

---

## As a DEVELOPER, I confirm that this PR **doesn't contain**:
- [x] Code lines with syntax errors
- [x] Methods and variables names that aren't clear or doesn't mean anything
- [x] Code that is out of the standard (Example: **PEP-8** for Python code)
- [x] Files with **imports** errors or without any utility
- [x] **TODOs** left behind
- [x] Tests that are being skipped
- [x] Methods placed on incorrect files or folders (Example: methods that are local, but should be placed on /utils)
- [x] Functions, methods or API responses that are publicly returning sensitive information

## As a DEVELOPER, I confirm that I:
- [x] Followed the Simbiose's standard for branch and commit naming
- [x] Concluded the task and achieved the goals according to the task planning I defined before starting to code
- [x] Added all necessary database changes to the [SQL Migrations](https://github.com/s1mbi0se/sql-migrations)
- [x] Created tests for every new flow and method on this Pull Request
- [x] Added the necessary DEBUG logs
- [x] Checked my code using the Simbiose's Corner Cases Guide
- [x] Merged the **develop** branch to this PR/branch before sending the PR for review
- [x] Solved all possible merge conflicts before sending the PR for review
- [x] Corrected all the problems and code smells identified by the quality tools
- [x] Manually executed the necessary tests locally to make sure my changes didn't break anything
- [x] All unitary and integration tests ran by the CI system are passing
- [x] Check the performance of my changes or additions and they didn't make the system slower (I profiled the code)
- [x] Updated the necessary documentation (Example: OpenAPI, Docstrings, Metadata Database Modeling)
- [x] Made the self-review of the code before sending the PR for review
- [x] Sent this PR for a peer-review
- [x] Made all the corrections appointed during the peer-review
- [x] Confirmed that this PR merge will not cause other systems or dependencies to stop working (Example: ShannonDB Python Client and ShannonDB)
- [x] Checked all GitHub Actions related to this PR that can block my merge

---  

## As the TEAM MANAGER, I confirm that I:
- [x] Checked that the code was developed following the best practices for the language, with the best possible readability and using the less possible quantity of code lines and complexity. 
- [x] Checked if the Pull Request and commit messages follow the Simbiose’s defined standard 
- [x] Checked if there is no possible merge conflicts before reviewing the PR
- [x] Checked if the purpose of the task was reached with this PR
- [x] Checked that the code was developed following the best practices for the language, with the best possible readability and using the less possible quantity of code lines and complexity. 
- [x] Checked if all related PRs were created when other components were affected by this task (e.g. database changes on [SQL Migrations](https://github.com/s1mbi0se/sql-migrations))
- [x] Checked that were created unit, integration and chaos tests for every new flow and method on this Pull Request
- [x] Checked that no sensible information (e.g. users security data, internal security data, etc) was exposed by this PR
- [x] Checked that code coverage for this PR is higher than 80%.
- [x] Checked that there are no problems and code smells identified by the quality and security tools
- [x] Checked that all files were created on the proper folders according to the Simbiose’s defined standard
- [x] Checked that all classes, methods and variables on this PR follow the Simbiose’s pattern for this programming language
- [x] Checked that all unitary and integration tests ran by the CI system are passing
- [x] Checked that there are no performance issues created by these changes or additions
- [x] Checked that all documentation is up to date according to this change (Example: OpenAPI, Docstrings, Metadata Database Modeling)
- [x] Checked that this PR merge will not cause other systems or dependencies to stop working (Example: ShannonDB Python Client and ShannonDB)
- [x] Checked all GitHub Actions related to this PR that can block the merge

---  

## As the QUALITY RESPONSIBLE, I confirm that I:
- [x] Checked if the Pull Request and commit messages follow the Simbiose’s defined standard 
- [x] Checked if there is no possible merge conflicts before reviewing the PR
- [x] Checked that were created unit, integration and chaos tests for every new flow and method on this Pull Request
- [x] Checked that no sensible information (e.g. users security data, internal security data, etc) was exposed by this PR
- [x] Checked that code coverage for this PR is higher than 80%.
- [x] Checked that there are no problems and code smells identified by the quality and security tools
- [x] Checked that all files were created on the proper folders according to the Simbiose’s defined standard
- [x] Checked that all classes, methods and variables on this PR follow the Simbiose’s code standard for this programming language
- [x] Checked that all unitary and integration tests ran by the CI system are passing
- [x] Checked that there are no performance issues created by these changes or additions
- [x] Checked that all documentation is up to date according to this change (Example: OpenAPI, Docstrings, Metadata Database Modeling)
- [x] Checked that this PR merge will not cause other systems or dependencies to stop working (Example: ShannonDB Python Client and ShannonDB)
- [x] Checked all GitHub Actions related to this PR that can block the merge

## Important links:
1. [Simbiose Quality Guidelines](https://github.com/s1mbi0se/company/wiki/Quality-Topics#table-of-contents)
2. [Metadata Database Modeling](https://docs.google.com/spreadsheets/d/1Y8_k5xV86kTjgfjbi-PGsJcubUKhWlT2x8ecpnMlyR0/edit?usp=sharing)
3. [Simbiose's Corner Cases Guide](https://www.simbioseventures.com/documentos-e-guias/procedimentos-e-checklists/corner-cases-guide)

### CODE REVIEW

File: core/base/src/main/java/alluxio/security/authorization/AccessControlList.java [hunk 1]
```diff
@@ -86,6 +86,9 @@ public AccessControlList() {
 
   /**
    * Clears out all entries (does not modify the owner name and owning group).
+   * <p>
+   * Sets the ACL {@link #mMode} to {@code 0} and the {@link #mExtendedEntries}
```
Comment (caiopavanelli) [inline core/base/src/main/java/alluxio/security/authorization/AccessControlList.java]: Unnecessary links to `mMode` and `mExtendedEntries`. Try to explain with words the result of the assignments of these variables without referencing implementation. What clearing the entries represent? No access? Full access? Try to find usages to figure this out.

File: core/base/src/main/java/alluxio/security/authorization/DefaultAccessControlList.java [hunk 2]
```diff
@@ -166,15 +166,16 @@ public void setEntry(AclEntry entry) {
   }
 
   /**
-   * Returns true if the default ACL is empty.
-   * @return true if empty
+   * @return whether this access-control list is empty
+   *         ({@code true}) or not ({@code false})
```
Comment (caiopavanelli) [inline core/base/src/main/java/alluxio/security/authorization/DefaultAccessControlList.java]: Thes second line is redundant as the return type is a `boolean`, so it's either true of false.

File: core/common/src/main/java/alluxio/wire/FileInfo.java [hunk 3]
```diff
@@ -150,7 +150,8 @@ public boolean isCompleted() {
   }
 
   /**
-   * @return whether the file is a folder
+   * @return whether the file is a folder ({@code true})
```
Comment (caiopavanelli) [inline core/common/src/main/java/alluxio/wire/FileInfo.java]: I feel it's unnecessary to tell the boolean can be `true` or `false`.

File: core/common/src/main/java/alluxio/wire/FileInfo.java [hunk 4]
```diff
@@ -256,7 +259,7 @@ public int getMode() {
   }
 
   /**
-   * @return the file persistence state
+   * @return the file {@link #mPersistenceState}
```
Comment (caiopavanelli) [inline core/common/src/main/java/alluxio/wire/FileInfo.java]: don't bind the docs with code variables.

File: core/common/src/main/java/alluxio/wire/FileInfo.java [hunk 5]
```diff
@@ -358,8 +361,11 @@ public DefaultAccessControlList getDefaultAcl() {
   }
 
   /**
-   * @param fileId the file id to use
-   * @return the file information
+   * Sets the ID for this file information.
+   *
+   * @param fileId the file ID to use
+   * @return this {@link FileInfo} updated
+   *         with a new {@link #mFileId}
```
Comment (caiopavanelli) [inline core/common/src/main/java/alluxio/wire/FileInfo.java]: don't bind the docs with code variables.

File: core/common/src/main/java/alluxio/wire/FileInfo.java [hunk 6]
```diff
@@ -376,8 +382,11 @@ public FileInfo setFileIdentifier(String fileIdentifier) {
   }
 
   /**
+   * Sets the name for this file information.
+   *
    * @param name the file name to use
-   * @return the file information
+   * @return this {@link FileInfo} object updated
+   *         with a new {@link #mName}
```
Comment (caiopavanelli) [inline core/common/src/main/java/alluxio/wire/FileInfo.java]: don't bind the docs with code variables.

File: core/common/src/main/java/alluxio/wire/FileInfo.java [hunk 7]
```diff
@@ -386,8 +395,13 @@ public FileInfo setName(String name) {
...
   /**
-   * @param path the file path to use
-   * @return the file information
+   * Sets the path for this file information.
+   *
+   * @param path the new file path to use for the
+   *        file this object holds information
+   *        from
+   * @return this {@link FileInfo} object updated
+   *         with a new {@link #mPath}
```
Comment (caiopavanelli) [inline core/common/src/main/java/alluxio/wire/FileInfo.java]: don't bind the docs with code variables.

File: core/common/src/main/java/alluxio/wire/FileInfo.java [hunk 8]
```diff
@@ -396,8 +410,11 @@ public FileInfo setPath(String path) {
   }
 
   /**
-   * @param ufsPath the file UFS path to use
-   * @return the file information
+   * Sets the UFS path for this file information.
+   *
+   * @param ufsPath the new file UFS path to use for this file information
+   * @return this file information updated with the provided {@code ufsPath}
+   *         as the new {@link #mUfsPath}
```
Comment (caiopavanelli) [inline core/common/src/main/java/alluxio/wire/FileInfo.java]: don't bind the docs with code variables.

File: core/common/src/main/java/alluxio/wire/FileInfo.java [hunk 9]
```diff
@@ -434,43 +460,61 @@ public FileInfo setCreationTimeMs(long creationTimeMs) {
...
     mCompleted = completed;
     return this;
   }
 
   /**
-   * @param folder the folder flag value to use
-   * @return the file information
+   * @param folder the folder flag value to use,
+   *        indicating whether the file is
+   *        a folder ({@code true}) or not ({@code false})
```
Comment (caiopavanelli) [inline core/common/src/main/java/alluxio/wire/FileInfo.java]:
```suggestion
   *        a folder or not
```

File: core/common/src/main/java/alluxio/wire/FileInfo.java [hunk 10]
```diff
@@ -434,43 +460,61 @@ public FileInfo setCreationTimeMs(long creationTimeMs) {
...
   public FileInfo setFolder(boolean folder) {
     mFolder = folder;
     return this;
   }
 
   /**
-   * @param pinned the pinned flag value to use
-   * @return the file information
+   * @param pinned the pinned flag value to use,
+   *        indicating whether the file is pinned
```
Comment (caiopavanelli) [inline core/common/src/main/java/alluxio/wire/FileInfo.java]:
```suggestion
   *        indicating whether the file is pinned or not
```

File: core/common/src/main/java/alluxio/wire/FileInfo.java [hunk 11]
```diff
@@ -434,43 +460,61 @@ public FileInfo setCreationTimeMs(long creationTimeMs) {
...
+   *        ({@code true}) or not ({@code false})
```
Comment (caiopavanelli) [inline core/common/src/main/java/alluxio/wire/FileInfo.java]:
```suggestion

```

File: core/common/src/main/java/alluxio/wire/FileInfo.java [hunk 12]
```diff
@@ -434,43 +460,61 @@ public FileInfo setCreationTimeMs(long creationTimeMs) {
...
     mPinned = pinned;
     return this;
   }
 
   /**
-   * @param cacheable the cacheable flag value to use
-   * @return the file information
+   * @param cacheable the cacheable flag value to use,
+   *        which indicates whether the file
+   *        is cacheable ({@code true}) or not
```
Comment (caiopavanelli) [inline core/common/src/main/java/alluxio/wire/FileInfo.java]:
```suggestion
   *        is cacheable or not
```

File: core/common/src/main/java/alluxio/wire/FileInfo.java [hunk 13]
```diff
@@ -434,43 +460,61 @@ public FileInfo setCreationTimeMs(long creationTimeMs) {
...
     mCacheable = cacheable;
     return this;
   }
 
   /**
-   * @param persisted the persisted flag value to use
-   * @return the file information
+   * @param persisted the persisted flag value to use,
+   *        which indicates whether the file
+   *        should be persisted ({@code true}) or not
```
Comment (caiopavanelli) [inline core/common/src/main/java/alluxio/wire/FileInfo.java]: don't need to tell the possible values of boolean.

File: core/common/src/main/java/alluxio/wire/FileInfo.java [hunk 14]
```diff
@@ -581,8 +635,11 @@ public FileInfo setPersistenceState(String persistenceState) {
   }
 
   /**
-   * @param mountPoint the mount point flag value to use
-   * @return the file information
+   * @param mountPoint the mount point flag value to use,
+   *        which indicates whether the file is a mount point
+   *        ({@code true}) or not ({@code false})
```
Comment (caiopavanelli) [inline core/common/src/main/java/alluxio/wire/FileInfo.java]: don't need to tell the possible values of boolean.

File: shell/src/main/java/alluxio/cli/fs/command/LsCommand.java [hunk 15]
```diff
@@ -156,19 +156,23 @@
   /**
    * Formats the ls result string.
    *
-   * @param hSize print human-readable format sizes
-   * @param acl whether security is enabled
-   * @param isFolder whether this path is a file or a folder
+   * @param hSize whether to print human-readable format sizes
+   *        ({@code true}) or not ({@code false})
```
Comment (caiopavanelli) [inline shell/src/main/java/alluxio/cli/fs/command/LsCommand.java]: Don't need to tell the possible values of boolean. Check all the parameters.

File: shell/src/main/java/alluxio/cli/fs/command/LsCommand.java [hunk 16]
```diff
@@ -195,6 +199,15 @@ public static String formatLsString(boolean hSize, boolean acl, boolean isFolder
     }
   }
 
+  /**
+   * Prints out the output for the file system command ls.
+   *
+   * @param status the URI status, used to determine whether there are extended
+   *        access-control lists (ACLs)
+   * @param hSize whether to print file/folder sizes in human-readable format
+   *        ({@code true}) or not ({@code false})
```
Comment (caiopavanelli) [inline shell/src/main/java/alluxio/cli/fs/command/LsCommand.java]: don't need to tell the possible values of boolean.

File: shell/src/main/java/alluxio/cli/fs/command/LsCommand.java [hunk 17]
```diff
@@ -296,6 +309,14 @@ public void run() {
     }
   }
 
+  /**
+   * Returns a version of the provided list of URI statuses sorted by field.
+   *
+   * @param statuses list of URI statuses to be sorted
+   * @param sortField string corresponding to the sort field comparator to use
+   * @param reverse whether sorting should be reversed ({@code true}) or not ({@code false})
```
Comment (caiopavanelli) [inline shell/src/main/java/alluxio/cli/fs/command/LsCommand.java]: don't need to tell the possible values of boolean.

**Review by caiopavanelli — CHANGES_REQUESTED** [review]

File: core/common/src/main/java/alluxio/wire/FileInfo.java (hunk 3 above)
Comment (caiopavanelli) [reply core/common/src/main/java/alluxio/wire/FileInfo.java]: check

File: shell/src/main/java/alluxio/cli/fs/command/LsCommand.java (hunk 15 above)
Comment (caiopavanelli) [reply shell/src/main/java/alluxio/cli/fs/command/LsCommand.java]: check

File: shell/src/main/java/alluxio/cli/fs/command/LsCommand.java (hunk 16 above)
Comment (caiopavanelli) [reply shell/src/main/java/alluxio/cli/fs/command/LsCommand.java]: check

File: shell/src/main/java/alluxio/cli/fs/command/LsCommand.java (hunk 17 above)
Comment (caiopavanelli) [reply shell/src/main/java/alluxio/cli/fs/command/LsCommand.java]: check

**Review by caiopavanelli — CHANGES_REQUESTED** [review]

**Review by caiopavanelli — APPROVED** [review]

File: core/client/fs/src/main/java/alluxio/util/FileSystemOptions.java [hunk 18]
```diff
@@ -173,8 +173,10 @@ public static GetStatusPOptions getStatusDefaults(AlluxioConfiguration conf) {
   }
 
   /**
+   * Returns the default list status options for the provided configuration.
+   *
    * @param conf Alluxio configuration
-   * @return options based on the configuration
+   * @return default options based on the {@code conf}iguration
```
Comment (jpedroantunes) [inline core/client/fs/src/main/java/alluxio/util/FileSystemOptions.java]: Is this text right? This "{@code conf}iguration" seems strange

**Review by jpedroantunes — CHANGES_REQUESTED** [review]

**Review by jpedroantunes — APPROVED** [review]
